src/util.cpp

```diff
@@ -444,6 +456,24 @@ void ArgsManager::ParseParameters(int argc, const char* const argv[])
         } else {
             m_override_args[key].push_back(val);
         }
+
+        // Check that the arg is known
+        bool found = false;
+        for (auto& arg : m_available_args) {
+            std::string arg_name = arg.first.second;
+            size_t eq_index = arg_name.find('=');
+            if (eq_index != std::string::npos) {
+                arg_name.erase(eq_index);
+            }
+            if (arg_name == ArgsManagerHelper::GetArgWithoutNet(key) || (IsSwitchChar(key[0]) && key.size() == 1)) {
```

Comment:
Maybe I'm missing something but I don't understand the second condition for 2 reasons:
 1. could be out of the loop since it doesn't depend on `arg`;
 2. `key.size() == 1` means that `key` is just `-`.

Reply:
Yeah, we need to ignore a key of `-` for bitcoin-cli. I moved the check for that outside of the loop.